site/docs/skylark/language.md

```diff
@@ -37,142 +29,97 @@ def fizz_buzz(n):
-least the 1.x release lifecycle. This ensures that Python-based tooling can at
-least parse Starlark code. Although Starlark is not *semantically* a subset of
-Python, behavioral differences are rare (excluding cases where Starlark raises
-an error).
+Starlark's semantics can differ from Python, and behavioral differences are rare, except for cases where Starlark raises an error. The following Python types are supported:
 
+* [None](lib/globals.html#None)
+* [bool](lib/bool.html)
+* [dict](lib/dict.html)
+* function
+* [int](lib/int.html)
+* [list](lib/list.html)
+* [string](lib/string.html)
 
 ## Mutability
 
-Because evaluation of BUILD and .bzl files is performed in parallel, there are
-some restrictions in order to guarantee thread-safety and determinism. Two
-mutable data structures are available: [lists](lib/list.html) and
-[dicts](lib/dict.html).
+Starlark favors immutability. Two mutable data structures are available: [lists](lib/list.html) and [dicts](lib/dict.html). Changes to mutable data-structures, such as appending a value to a list or deleting an entry in a dictionary are valid only for objects created in the current context. After a context finishes, its values become immutable.
 
-In a build, there are many "evaluation contexts": each `.bzl` file and each
-`BUILD` file is loaded in a different context. Each rule is also analyzed in a
-separate context. We allow side-effects (e.g. appending a value to a list or
-deleting an entry in a dictionary) only on objects created during the current
-evaluation context. Once the code in that context is done executing, all of its
-values are frozen.
+This is because Bazel builds use parallel execution. During a build, each `.bzl` file and each `BUILD` file get their own execution context. Each rule is also analyzed in its own context.
 
-For example, here is the content of the file `foo.bzl`:
+Let's go through an example with the file `foo.bzl`:
 
 ```python
-var = []
+# `foo.bzl`
+var = [] # declare a list
 
-def fct():
-  var.append(5)
+def fct(): # declare a function
+  var.append(5) # append a value to the list
 
-fct()
+fct() # execute the fct function
 ```
 
-The variable `var` is created when `foo.bzl` is loaded. `fct()` is called during
-the same context, so it is safe. At the end of the evaluation, the environment
-contains an entry mapping the identifier `var` to a list `[5]`; this list is
-then frozen.
+Bazel creates `var` when `foo.bzl` loads. `var` is thus part of `foo.bzl`'s context. When `fct()` runs, it does so within the context of `foo.bzl`. After evaluation for `foo.bzl` completes, the environment contains an immutable entry, `var`, with the value `[5]`.
 
-It is possible for multiple other files to load symbols from `foo.bzl` at the
-same time. For this reason, the following code is not legal:
+When another, `bar.bzl` loads symbols from `foo.bzl`, loaded values remain immutable. For this reason, the following code in `bar.bzl` is illegal:
```

Comment:
Nit: "When another," -> "When another" (delete comma)

Reply:
Thank you! deleted it!